### PR TITLE
Refactor/typescript

### PR DESCRIPTION
**Pull request by DeveloperKhan**
Rewrote codebase in TS
- Setup a TS config using Next.js (https://nextjs.org/docs/basic-features/typescript)
- Converted all .js to either .ts or .tsx
- Removed prop-types, replaced with interfaces
- Added additional config to prevent errors with SCSS Modules

Note: It may be more efficient to use "type" instead of "interface" at some parts, but IMO the codebase is small enough where "type" would only be useful for a very time set of cases

### CODE REVIEW

File: components/shared/Team/index.tsx [hunk 1]
```diff
@@ -8,7 +7,12 @@ import Item from './team-item';
 import styles from './styles.module.scss';
 import person from './item-interface';
 
-function Team({ infinite, items }) {
+interface Props {
+  items: [person];
```
Comment (ghost) [inline components/shared/Team/index.tsx]: This isn't _quite_ the same thing as an array of person; it's actually an array with 1 item of the Person type. Use `Person[]` instead

File: components/shared/Project/item-interface.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+interface project {
```
Comment (ghost) [inline components/shared/Project/item-interface.ts]: Def capitalize interface names

File: components/shared/Team/item-interface.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+interface person {
```
Comment (ghost) [inline components/shared/Team/item-interface.ts]: Def capitalize interface names

File: next-env.d.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+/// <reference types="next" />
```
Comment (ghost) [inline next-env.d.ts:1]: Should this file get committed?

**Review by ghost — CHANGES_REQUESTED** [review]
Some small adjustments but looks good! I also kinda prefer using the "type" instead of "interface;" a little cleaner and it doesn't remind me of Java 😁

File: next-env.d.ts (hunk 4 above)
Comment (DeveloperKhan) [reply next-env.d.ts:1]: it should, the docs say you shouldn't remove it and some of the settings in this file prevent eslint errors

File: components/shared/Team/index.tsx (hunk 1 above)
Comment (DeveloperKhan) [reply components/shared/Team/index.tsx]: fixed

File: components/shared/Project/item-interface.ts (hunk 2 above)
Comment (DeveloperKhan) [reply components/shared/Project/item-interface.ts]: fixed

File: components/shared/Team/item-interface.ts (hunk 3 above)
Comment (DeveloperKhan) [reply components/shared/Team/item-interface.ts]: fixed

File: next-env.d.ts (hunk 4 above)
Comment (ghost) [reply next-env.d.ts:1]: 👍

**Pull request merged**